lib/rules/lint-require-valid-alt-text.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const AstNodeInfo = require('../helpers/ast-node-info');
+const Rule = require('./base');
+
+function hasAccessibleChild(node) {
+  return AstNodeInfo.hasChildren(node);
+}
+
+module.exports = class A11yLintAltText extends Rule {
+  logNode({ node, message }) {
+    return this.log({
+      message,
+      line: node.loc && node.loc.start.line,
+      column: node.loc && node.loc.start.column,
+      source: this.sourceForNode(node),
+    });
+  }
+  visitor() {
+    return {
+      ElementNode(node) {
+        const isImg = AstNodeInfo.isImgElement(node);
+        const isObj = AstNodeInfo.isObjectElement(node);
+        const isInput = AstNodeInfo.isInputElement(node);
+        const isArea = AstNodeInfo.isAreaElement(node);
+        const hasAriaHiddenAttribute = AstNodeInfo.hasAttributeValue(node, 'aria-hidden', 'true');
+        if (hasAriaHiddenAttribute) {
+          return;
+        }
+        if (isImg) {
+          const hasAltAttribute = AstNodeInfo.hasAttribute(node, 'alt');
+          const altValue = AstNodeInfo.elementAttributeValue(node, 'alt');
+          const roleValue = AstNodeInfo.elementAttributeValue(node, 'role');
+          if (!hasAltAttribute) {
+            this.logNode({
+              message:
+                'img tags must have an alt attribute, either with meaningful text, or an empty string for decorative images.',
```

Comment:
img elements must have an alt attribute. See https://www.w3.org/WAI/WCAG21/Techniques/html/H37 for more details.